include/fiction/utils/layout_utils.hpp

```diff
@@ -350,7 +352,54 @@ Lyt convert_to_fiction_coordinates(const sidb_cell_clk_lyt_siqad& lyt) noexcept
 
     return lyt_new;
 }
+/**
+ * Generates a random coordinate within the region spanned by two given coordinates. The two given coordinates form the
+ * lower left corner and the upper right corner of the spanned region.
+ *
+ * @OffsetCoordinateType The coordinate implementation to be used.
+ * @param coordinate1 Coordinate of the lower left corner.
+ * @param coordinate2 Coordinate of the upper right corner (coordinate order is not important, automatically swapped if
+ * necessary).
+ * return Randomly generated coordinate.
+ */
+template <typename CoordinateType>
+CoordinateType random_coordinate(const CoordinateType& coordinate1, const CoordinateType& coordinate2) noexcept
+{
+    static std::mt19937_64 generator(std::random_device{}());
+
+    CoordinateType left_coordinate  = coordinate1;
+    CoordinateType right_coordinate = coordinate2;
+
+    if (left_coordinate > right_coordinate)
+    {
+        std::swap(left_coordinate, right_coordinate);
+    }
+
+    if constexpr (std::is_same_v<CoordinateType, offset::ucoord_t>)
+    {
+        std::uniform_int_distribution<> dist_x(left_coordinate.x, right_coordinate.x);
+        std::uniform_int_distribution<> dist_y(left_coordinate.y, right_coordinate.y);
+        std::uniform_int_distribution<> dist_z(left_coordinate.z, right_coordinate.z);
 
+        return {dist_x(generator), dist_y(generator), dist_z(generator)};
+    }
+    else if constexpr (std::is_same_v<CoordinateType, cube::coord_t>)
+    {
+        std::uniform_int_distribution<> dist_x(left_coordinate.x, right_coordinate.x);
+        std::uniform_int_distribution<> dist_y(left_coordinate.y, right_coordinate.y);
+        std::uniform_int_distribution<> dist_z(left_coordinate.z, right_coordinate.z);
+
+        return {dist_x(generator), dist_y(generator), dist_z(generator)};
+    }
+    else if constexpr (std::is_same_v<CoordinateType, siqad::coord_t>)
+    {
+        std::uniform_int_distribution<> dist_x(left_coordinate.x, right_coordinate.x);
+        std::uniform_int_distribution<> dist_y(left_coordinate.y, right_coordinate.y);
+        std::uniform_int_distribution<> dist_z(left_coordinate.z, right_coordinate.z);
+
+        return {dist_x(generator), dist_y(generator), dist_z(generator)};
+    }
```

Comment:
Good point! They just got the same after I removed the integer type in std::uniform_int_distribution<> (I was having trouble passing all the checks in the workflow). Should be fine since all the boundaries are determined by the coordinates themselves, right?

Reply:
The coordinates have intrinsic min/max values, indeed.